runner/src/main/kotlin/pl/edu/prz/kod/runner/adapters/http/ErrorHandler.kt

```diff
@@ -2,33 +2,47 @@ package pl.edu.prz.kod.runner.adapters.http
 
 import org.http4k.core.Response
 import org.http4k.core.Status
+import org.http4k.format.Jackson
+import pl.edu.prz.kod.common.adapters.http.dto.ErrorResponse
 import pl.edu.prz.kod.runner.adapters.http.dto.DecodingResult
 import pl.edu.prz.kod.runner.domain.ExecutionResult
 
-fun handleDecodingError(result: DecodingResult.Failure): Response = when (result) {
-    is DecodingResult.Failure.LanguageNotImplementedResult -> {
-        logEvent(
-            LanguageNotImplementedEvent(result.language)
-        )
-        languageNotImplementedResponse(result.language)
-    }
-}
+class ErrorHandler {
+    companion object {
```

Comment:
How about creating an instance of ErrorHandler using KOIN and injecting it into HttpHandler rather than using companion object with JVMStatic? I guess it would make our code more testable in the future.